### ISSUE TITLE
uv does not recognize __main__.py?

### ISSUE THREAD

**Issue by seppzer0** [astral-sh/uv#7275]
I have a Python project that I run directly from source. Dependencies are stored within a uv-managed .venv in the project directory.

When I attempt to launch it as I normally would with `PYTHONPATH=$(pwd) python3 builder/` via my system installation, I get an error:

```txt
$ PYTHONPATH=$(pwd) uv run ./builder
error: Failed to spawn: `./builder`
  Caused by: Access is denied. (os error 5)
```

However, if I just add the `__main__.py` in my command:

```txt
$ PYTHONPATH=$(pwd) uv run ./builder/__main__.py 
usage: __main__.py [-h] [--clean] {kernel,assets,bundle} ...

A custom builder for the zero kernel.

positional arguments:
  {kernel,assets,bundle}
    kernel              build the kernel
    assets              collect assets
    bundle              build the kernel + collect assets

options:
  -h, --help            show this help message and exit
  --clean               clean the root directory
```

..it runs successfully.

Would this be an intended behaviour or is this a bug?

**Comment by charliermarsh** [comment]
Does `uv run python ./builder` work as expected?

**Comment by seppzer0** [comment]
Interestingly, it does. I don't understand the difference yet, but adding `python` works.

**Comment by zanieb** [comment]
In the error case, we've attempted to run a directory as though it were an executable. In the latter, you've invoked Python to run the module.

We should probably special case directories with `__main__.py` though (and support what you attempted to do)

**Comment by ghpqans** [comment]
Same problem here. Feels uncomfortable to enter sth like `uv run src/projectname/__main__.py` . Script section in `pyproject.toml` also does not support `__main.py__`.

**Comment by konstin** [comment]
@ghpqans This issue has been solved, please open a new issue following #9452.
